### PR TITLE
Make the FigureBuilder more fail tolerant

### PR DESCRIPTION
**Pull request by m-vo**
| Q                | A
| -----------------| ---
| Fixed issues     | Partly solves #2709 
| Docs PR or issue | todo

This PR addresses https://github.com/contao/contao/issues/2709#issuecomment-777626885 (1.). There now is an alternative `buildIfResourceExists()` and throwing of `InvalidResourceExceptions` is deferred to the `build()` method.

**Todo:**
 - [x] Adjust usage after #2732 is merged upstream

### CODE REVIEW

**Comment by m-vo** [comment]
This one is rebased + imo ready to go.

**Review by leofeyer — DISMISSED** [review]
LGTM. 👍🏻 Waiting for @ausi's approval.

File: core-bundle/tests/Image/Studio/FigureBuilderTest.php
Comment (ausi) [inline core-bundle/tests/Image/Studio/FigureBuilderTest.php]:
```suggestion
        foreach ($setInvalidResourceOperations as [$method, $argument]) {
```

File: core-bundle/tests/Image/Studio/FigureBuilderTest.php
Comment (ausi) [inline core-bundle/tests/Image/Studio/FigureBuilderTest.php]:
```suggestion
        foreach ($setValidResourceOperations as [$method, $argument]) {
```

**Review by ausi — DISMISSED** [review]
LGTM 👍
_(suggested code changes are optional ☺️)_

File: core-bundle/tests/Image/Studio/FigureBuilderTest.php
Comment (m-vo) [reply core-bundle/tests/Image/Studio/FigureBuilderTest.php]: short and expressive - I like it :slightly_smiling_face:

Comment (leofeyer) [reply core-bundle/tests/Image/Studio/FigureBuilderTest.php]: I didn't even know this was possible. 🦊

**Review by ausi — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @m-vo.

**Pull request merged**
